### PR TITLE
feat(qa-agent): reformulate prompt to adversarial product tester (B-lean)

### LINKED ISSUE #135: improvement: reformulate QA Agent prompt to adversarial product tester
## Problem

The QA Agent uses a confirmation posture — it asks "were the ACs implemented?" rather than "what is missing?". Observed in PR #133: Sentinel caught a discrepancy between the local hook and the CI workflow that the QA Agent missed. The agent provides weak signal because it confirms rather than challenges.

## Sprint Goal / Success Metrics

| Metric | Baseline | Target | When |
|--------|----------|--------|------|
| QA Agent identifies AC gaps (not just coverage) | 0% of PRs flagged for gaps | >50% of PRs with real gaps get flagged | After deploy |
| No overlap with Sentinel (code/file/tech reviews) | N/A | 0 QA comments about code quality | After deploy |

## Solution

Reformulate the QA Agent prompt from "senior QA engineer confirming AC coverage" to "adversarial product tester finding gaps." Output format changes to B-lean: `PASS|FAIL` verdict on first line, then `Gaps:` (undefined edge cases, ambiguous states, missing scenarios) and `Not covered:` (AC references where diff falls short) — one phrase each, or "none"/"all covered" when clean. The agent is explicitly instructed not to review code quality, missing files, or technical consistency — that boundary belongs to Sentinel.

## Acceptance Criteria

**AC1 — Adversarial posture**
- Given a PR whose ACs have undefined edge cases or ambiguous states
- When QA Agent runs
- Then it reports gaps in the ACs themselves, not just whether the diff implements them

**AC2 — B-lean output format**
- Given any QA Agent run
- When it posts a PR comment
- Then the comment follows exactly: line 1 `PASS` or `FAIL`, line 2 `Gaps: [one-line summary or "none"]`, line 3 `Not covered: [AC refs or "all covered"]`

**AC3 — Sentinel boundary preserved**
- Given a PR with missing files or technical inconsistencies
- When QA Agent runs
- Then it does NOT comment on code quality, file structure, or technical consistency

## Edge Cases

- EC1: No linked issue → agent uses PR description as AC source, same adversarial posture applies
- EC2: ACs are empty or minimal → agent flags "ACs insufficient to evaluate" under Gaps
- EC3: GitHub Models API error → existing `infra:qa-broken` label path unchanged

## Out of Scope

- Changing the model (gpt-4o-mini) — not part of this issue
- Changing label automation (`qa:approved` / `qa:needs-work`) — labels and triggers unchanged
- Sentinel prompt changes — separate concern

## Non-Functional Requirements

- Performance: same 30s timeout, same single API call structure
- Security: no new secrets required
- Reliability: API error fallback path unchanged; PASS/FAIL parsing logic unchanged

## Files to Modify

- `templates/.github/workflows/qa-agent.yml` — reformulate `PROMPT_JSON` string (~line 54); update comment body interpolation to render B-lean format

### PR DESCRIPTION
**Pull request by rafaeltcosta86**
## Summary

- Replaces `"senior QA engineer"` confirmation posture with `"adversarial product tester"` — agent now finds AC gaps, not just confirms coverage
- Output format changed to B-lean: `PASS|FAIL` + `Gaps:` + `Not covered:` (compact, structured, no overlap with Sentinel)
- Explicit prompt boundary: QA Agent must not review code quality, file structure, or technical consistency

## Test plan

- [ ] Open a PR with ACs that have undefined edge cases → verify QA Agent reports `Gaps:` with content
- [ ] Open a PR with complete ACs fully covered → verify `PASS` + `Gaps: none` + `Not covered: all covered`
- [ ] Verify comment format matches B-lean (3-line structure under the verdict)
- [ ] Verify Sentinel-territory issues (missing files, tech inconsistency) do NOT appear in QA comment

Closes #135

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: templates/.github/workflows/qa-agent.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@ jobs:
           fi
 
           # Serialize prompt as JSON string and call GitHub Models API (30s timeout)
-          PROMPT_JSON=$(printf '%s' "You are a senior QA engineer. Review whether this PR diff satisfies the Acceptance Criteria below.\n\nACCEPTANCE CRITERIA:\n${AC_CONTEXT}\n\nPR DIFF:\n${DIFF}\n\nFirst line of your response must be exactly one word: PASS or FAIL. Second line: brief explanation (max 3 sentences)." | python3 -c 'import json,sys; print(json.dumps(sys.stdin.read()))')
+          PROMPT_JSON=$(printf '%s' "You are an adversarial product tester. Your mission is to find what the Acceptance Criteria do NOT cover — undefined edge cases, ambiguous states, missing user scenarios. Do NOT review code quality, file structure, or technical consistency.\n\nACCEPTANCE CRITERIA:\n${AC_CONTEXT}\n\nPR DIFF:\n${DIFF}\n\nRespond in exactly 3 lines:\nLine 1: PASS or FAIL\nLine 2: Gaps: [one-line summary of AC gaps found, or \"none\"]\nLine 3: Not covered: [AC refs where diff falls short, or \"all covered\"]" | python3 -c 'import json,sys; print(json.dumps(sys.stdin.read()))')
```
[the remark this answers is not in the document]
Comment (rafaeltcosta86) [reply templates/.github/workflows/qa-agent.yml]: Fixed in 598940c (3 changes applied):

1. **Explicit PASS/FAIL criteria** — added `(PASS if the PR diff fully satisfies the stated Acceptance Criteria, FAIL if it falls short of covering them)` to Line 1 instruction, preventing adversarial framing from failing PRs that satisfy the stated ACs.

2. **No-markdown instruction** — added `do NOT wrap your response in markdown code blocks or any other formatting` to prevent fence-wrapped responses that break the parser.

3. **Parser fence strip** — beyond the prompt instruction, hardened the parser itself: `lines=[l for l in t.split("\n") if not l.strip().startswith("\`\`\`")]` strips fence lines before extracting the verdict line. Defense in depth — prompt instruction is first layer, parser strip is second.

**Tested:** parser unit tests show 4 previously-failing fence cases now pass, 0 regressions. Prompt tested against GitHub Models API — response format correct (3 lines, no fences), verdict parsed cleanly.
